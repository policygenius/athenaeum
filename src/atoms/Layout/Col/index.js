import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import colors from 'atoms/Color/colors.scss';
import styles from '../layout.module.scss';

function Col(props) {
  const {
    children,
    className,
    smallCols,
    mediumCols,
    largeCols,
    xLargeCols,
    xxLargeCols,
    padding,
    fullwidth,
    flex,
    style,
    bottomSpacing,
    onClick,
    borderColor
  } = props;

  const classes = [
    styles['col'],
    smallCols && styles[`col-sm-${smallCols}`],
    mediumCols && styles[`col-md-${mediumCols}`],
    largeCols && styles[`col-lg-${largeCols}`],
    xLargeCols && styles[`col-xlg-${xLargeCols}`],
    xxLargeCols && styles[`col-xxlg-${xxLargeCols}`],
    fullwidth && styles['fullwidth'],
    flex && styles['flex'],
    bottomSpacing && styles[`bottom-spacing-${bottomSpacing}`],
    padding && styles['padding'],
    colors[`border-${borderColor}`],
    className
  ];

  return (
    <div className={classnames(...classes)} style={style} onClick={onClick}>
      { children }
    </div>
  );
}

Col.propTypes = {
  /**
   * Supply any additional class names.
   */
  className: PropTypes.string,

  /**
   * sets bottom spacing between children in Layout
   * - see [Spacer](#spacer) for appropriate values
   */
  bottomSpacing: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),

  /**
   * Overides column grid width within a viewport from `0px` up. Width can also be set to 'auto'.
   */
  smallCols: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf([ 'auto' ])
  ]),

  /**
   * Sets col grid width in viewport from `768px` up. Width can also be set to 'auto'.
   */
  mediumCols: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf([ 'auto' ])
  ]),

  /**
   * Sets col grid width in viewport from `1024px` up. Width can also be set to 'auto'.
   */
  largeCols: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf([ 'auto' ])
  ]),

  /**
   * Sets col grid width in viewport from `1280px` up. Width can also be set to 'auto'.
   */
  xLargeCols: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf([ 'auto' ])
  ]),
  /**
   * Sets col grid width in viewport from `1440px` up. Width can also be set to 'auto'.
   */
  xxLargeCols: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf([ 'auto' ])
  ]),

  /**
   * Overides flex, changes Col to grow to fill empty space in the row (`flex-grow: 1`)
   */
  flex: PropTypes.bool,

  /**
   * Overides and removes padding set from Layout
   */
  fullwidth: PropTypes.bool,

  /**
   * Adds standard (~6px) left/right padding to the `Layout`
   */
  padding: PropTypes.bool,

  /**
   * adds additional styles to the column as a React styles object  ``
   * - For useful positional styles, [Checkout this Guide to Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)
   */
  style: PropTypes.object,
  /**
   * onClick handler
   */
  onClick: PropTypes.func,
  /**
   * Sets the border color for the Col. To see a border, give the border a width and a style directly.
   * Can only be one of the named brand colors.
   */
  borderColor: PropTypes.string
};

Col.defaultProps = {
  borderColor: 'neutral-6'
};

export default Col;
