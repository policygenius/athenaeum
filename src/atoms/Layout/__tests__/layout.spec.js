import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Layout from 'atoms/Layout';
import Col from 'atoms/Layout/Col';

describe('<Layout />', () => {

  it('creates a Layout component', () => {
    const wrapper = mount(<Layout />);

    expect(wrapper.type()).toEqual(Layout);
  });

  it('renders correctly', () => {
    const component = renderer.create(<Layout />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('can create variant via props', () => {
    const wrapper = shallow(<Layout variant='bordered-buckets' />);

    expect(wrapper.hasClass('bordered-buckets')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders its children', () => {
    const wrapper = shallow(
      <Layout smallCols={[ 1 ]}>
        <div className='child' />
        <div className='some-child' />
        <div className='child' />
      </Layout>
    );

    expect(wrapper.find('.child')).toHaveLength(2);
    expect(wrapper.find('.some-child')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('wraps its children in `<Col>`', () => {
    const wrapper = shallow(
      <Layout smallCols={[ 1 ]}>
        <div className='child' />
        <div className='child' />
        <div className='child' />
      </Layout>
    );

    expect(wrapper.find('Col')).toHaveLength(3);
    expect(wrapper).toMatchSnapshot();
  });

  it('doesn\'t wrap Cols in `<Col>`', () => {
    const wrapper = shallow(
      <Layout smallCols={[ 1 ]}>
        <div className='child' />
        <Col className='child' />
      </Layout>
    );

    expect(wrapper.find('Col > Col')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('passes its `col` props down to its `<Col>` correctly', () => {
    const wrapper = shallow(
      <Layout
        smallCols={[ 1 ]}
        mediumCols={[ 6 ]}
        largeCols={[ 7, 5 ]}
        bottomSpacing='xLarge'
        fullwidth
        padding
      >
        <div className='child' />
        <div className='child' />
      </Layout>
    );

    expect(wrapper.find('Col').first().props().smallCols).toBeTruthy();
    expect(wrapper.find('Col').first().props().smallCols).toEqual(1);

    expect(wrapper.find('Col').first().props().mediumCols).toBeTruthy();
    expect(wrapper.find('Col').first().props().mediumCols).toEqual(6);

    expect(wrapper.find('Col').first().props().fullwidth).toBeTruthy();
    expect(wrapper.find('Col').first().props().padding).not.toBeTruthy();
    expect(wrapper.find('Col').first().props().bottomSpacing).toEqual('xLarge');
    expect(wrapper).toMatchSnapshot();
  });

  it('wraps its child `<Col>` with column sizes', () => {
    const wrapper = shallow(
      <Layout
        largeCols={[ 7, 5 ]}
      >
        <div className='child' />
        <div className='child' />
      </Layout>
    );

    expect(wrapper.find('Col').get(0).props.largeCols).toEqual(7);
    expect(wrapper.find('Col').get(1).props.largeCols).toEqual(5);
    expect(wrapper).toMatchSnapshot();
  });

  it('wraps its child `<Col>` with column sizes', () => {
    const wrapper = shallow(
      <Layout
        smallCols={[ 1 ]}
      >
        <div className='child' />

        <Col smallCols={3}>
          <div className='one-off' />
        </Col>

        <div className='child' />

      </Layout>
    );

    expect(wrapper.children().get(0).props.smallCols).toEqual(1);
    expect(wrapper.children().get(1).props.smallCols).toEqual(3);
    expect(wrapper.children().get(2).props.smallCols).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

});
