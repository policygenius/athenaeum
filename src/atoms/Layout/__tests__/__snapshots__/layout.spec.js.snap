// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> can create variant via props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout bordered-buckets"
    id={undefined}
    style={undefined}
/>,
  "nodes": Array [
    <div
      className="layout bordered-buckets"
      id={undefined}
      style={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      largeCols={Array []}
      mediumCols={Array []}
      smallCols={
            Array [
                  12,
                ]
      }
      variant="bordered-buckets"
      xLargeCols={Array []}
      xxLargeCols={Array []}
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout bordered-buckets"
      id={undefined}
      style={undefined}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    largeCols={Array []}
    mediumCols={Array []}
    smallCols={
        Array [
            12,
          ]
    }
    variant="bordered-buckets"
    xLargeCols={Array []}
    xxLargeCols={Array []}
/>,
}
`;

exports[`<Layout /> doesn't wrap Cols in \`<Col>\` 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout"
    id={undefined}
    style={undefined}
>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        className="child"
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    />
</div>,
  "nodes": Array [
    <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            className="child"
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      largeCols={Array []}
      mediumCols={Array []}
      smallCols={
            Array [
                  1,
                ]
      }
      xLargeCols={Array []}
      xxLargeCols={Array []}
>
      <div
            className="child"
      />
      <Col
            borderColor="neutral-6"
            className="child"
      />
</Layout>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            className="child"
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    largeCols={Array []}
    mediumCols={Array []}
    smallCols={
        Array [
            1,
          ]
    }
    xLargeCols={Array []}
    xxLargeCols={Array []}
>
    <div
        className="child"
    />
    <Col
        borderColor="neutral-6"
        className="child"
    />
</Layout>,
}
`;

exports[`<Layout /> passes its \`col\` props down to its \`<Col>\` correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout padding"
    id={undefined}
    style={undefined}
>
    <Col
        borderColor="neutral-6"
        bottomSpacing="xLarge"
        flex={undefined}
        fullwidth={true}
        largeCols={7}
        mediumCols={6}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing="xLarge"
        flex={undefined}
        fullwidth={true}
        largeCols={5}
        mediumCols={6}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
</div>,
  "nodes": Array [
    <div
      className="layout padding"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing="xLarge"
            flex={undefined}
            fullwidth={true}
            largeCols={7}
            mediumCols={6}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing="xLarge"
            flex={undefined}
            fullwidth={true}
            largeCols={5}
            mediumCols={6}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      bottomSpacing="xLarge"
      fullwidth={true}
      largeCols={
            Array [
                  7,
                  5,
                ]
      }
      mediumCols={
            Array [
                  6,
                ]
      }
      padding={true}
      smallCols={
            Array [
                  1,
                ]
      }
      xLargeCols={Array []}
      xxLargeCols={Array []}
>
      <div
            className="child"
      />
      <div
            className="child"
      />
</Layout>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout padding"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing="xLarge"
            flex={undefined}
            fullwidth={true}
            largeCols={7}
            mediumCols={6}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing="xLarge"
            flex={undefined}
            fullwidth={true}
            largeCols={5}
            mediumCols={6}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    bottomSpacing="xLarge"
    fullwidth={true}
    largeCols={
        Array [
            7,
            5,
          ]
    }
    mediumCols={
        Array [
            6,
          ]
    }
    padding={true}
    smallCols={
        Array [
            1,
          ]
    }
    xLargeCols={Array []}
    xxLargeCols={Array []}
>
    <div
        className="child"
    />
    <div
        className="child"
    />
</Layout>,
}
`;

exports[`<Layout /> renders correctly 1`] = `
<div
  className="layout"
  id={undefined}
  style={undefined}
/>
`;

exports[`<Layout /> renders its children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout"
    id={undefined}
    style={undefined}
>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="some-child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
</div>,
  "nodes": Array [
    <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="some-child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      largeCols={Array []}
      mediumCols={Array []}
      smallCols={
            Array [
                  1,
                ]
      }
      xLargeCols={Array []}
      xxLargeCols={Array []}
>
      <div
            className="child"
      />
      <div
            className="some-child"
      />
      <div
            className="child"
      />
</Layout>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="some-child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    largeCols={Array []}
    mediumCols={Array []}
    smallCols={
        Array [
            1,
          ]
    }
    xLargeCols={Array []}
    xxLargeCols={Array []}
>
    <div
        className="child"
    />
    <div
        className="some-child"
    />
    <div
        className="child"
    />
</Layout>,
}
`;

exports[`<Layout /> wraps its child \`<Col>\` with column sizes 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout"
    id={undefined}
    style={undefined}
>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={7}
        mediumCols={undefined}
        smallCols={12}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={5}
        mediumCols={undefined}
        smallCols={12}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
</div>,
  "nodes": Array [
    <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={7}
            mediumCols={undefined}
            smallCols={12}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={5}
            mediumCols={undefined}
            smallCols={12}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      largeCols={
            Array [
                  7,
                  5,
                ]
      }
      mediumCols={Array []}
      smallCols={
            Array [
                  12,
                ]
      }
      xLargeCols={Array []}
      xxLargeCols={Array []}
>
      <div
            className="child"
      />
      <div
            className="child"
      />
</Layout>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={7}
            mediumCols={undefined}
            smallCols={12}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={5}
            mediumCols={undefined}
            smallCols={12}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    largeCols={
        Array [
            7,
            5,
          ]
    }
    mediumCols={Array []}
    smallCols={
        Array [
            12,
          ]
    }
    xLargeCols={Array []}
    xxLargeCols={Array []}
>
    <div
        className="child"
    />
    <div
        className="child"
    />
</Layout>,
}
`;

exports[`<Layout /> wraps its child \`<Col>\` with column sizes 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout"
    id={undefined}
    style={undefined}
>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={3}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="one-off"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
</div>,
  "nodes": Array [
    <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={3}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="one-off"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      largeCols={Array []}
      mediumCols={Array []}
      smallCols={
            Array [
                  1,
                ]
      }
      xLargeCols={Array []}
      xxLargeCols={Array []}
>
      <div
            className="child"
      />
      <Col
            borderColor="neutral-6"
            smallCols={3}
      >
            <div
                  className="one-off"
            />
      </Col>
      <div
            className="child"
      />
</Layout>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={3}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="one-off"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    largeCols={Array []}
    mediumCols={Array []}
    smallCols={
        Array [
            1,
          ]
    }
    xLargeCols={Array []}
    xxLargeCols={Array []}
>
    <div
        className="child"
    />
    <Col
        borderColor="neutral-6"
        smallCols={3}
    >
        <div
            className="one-off"
        />
    </Col>
    <div
        className="child"
    />
</Layout>,
}
`;

exports[`<Layout /> wraps its children in \`<Col>\` 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="layout"
    id={undefined}
    style={undefined}
>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
    <Col
        borderColor="neutral-6"
        bottomSpacing={undefined}
        flex={undefined}
        fullwidth={undefined}
        largeCols={undefined}
        mediumCols={undefined}
        smallCols={1}
        xLargeCols={undefined}
        xxLargeCols={undefined}
    >
        <div
            className="child"
        />
    </Col>
</div>,
  "nodes": Array [
    <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Layout
      largeCols={Array []}
      mediumCols={Array []}
      smallCols={
            Array [
                  1,
                ]
      }
      xLargeCols={Array []}
      xxLargeCols={Array []}
>
      <div
            className="child"
      />
      <div
            className="child"
      />
      <div
            className="child"
      />
</Layout>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="layout"
      id={undefined}
      style={undefined}
>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
      <Col
            borderColor="neutral-6"
            bottomSpacing={undefined}
            flex={undefined}
            fullwidth={undefined}
            largeCols={undefined}
            mediumCols={undefined}
            smallCols={1}
            xLargeCols={undefined}
            xxLargeCols={undefined}
      >
            <div
                  className="child"
            />
      </Col>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Layout
    largeCols={Array []}
    mediumCols={Array []}
    smallCols={
        Array [
            1,
          ]
    }
    xLargeCols={Array []}
    xxLargeCols={Array []}
>
    <div
        className="child"
    />
    <div
        className="child"
    />
    <div
        className="child"
    />
</Layout>,
}
`;
